sequence_processing_pipeline/Pipeline.py

```diff
@@ -238,12 +254,21 @@ def generate_sample_information_files(self):
         Generate sample-information files in self.output_path.
         :return: A list of paths to sample-information-files.
         """
-        samples = []
-        for sample in self.sample_sheet.samples:
-            if sample['Sample_ID'].startswith('BLANK'):
-                samples.append((sample['Sample_ID'], sample['Sample_Project']))
+        if self.mapping_file is not None:
+            # TODO: Note we may need to introduce a conversion to sample_id
+            #  format from sample_name.
+            df = self.mapping_file[['sample_name', 'project_name']]
+            samples = list(df.to_records(index=False))
```

Comment:
Note that you are basically looping twice over the same dataframe (`df`) in the next two lines; I don't think this is important now but good to know for when you work with large dataframes.

Reply:
Ty!